### PR TITLE
USB Gecko: give up after 5s timeout

### PR DESCRIPTION
**Pull request by webhdx**
This PR fixes one of the most annoying things in iplboot. When you have USB Gecko connected it will wait indefinitely for PC to accept data. This was problematic is all you wanted is to boot Swiss from SD card and use USB Gecko with Swiss. Now imagine testing 20 USB Geckos having to deal with this behavior :P 
This PR adds 5s window to send data to USB Gecko. I think this is enough time because most likely you already have usb-load waiting for connection when you start iplboot.

### CODE REVIEW

File: source/main.c [hunk 1]
```diff
@@ -256,9 +256,21 @@ int load_usb(char slot)
...
-    kprintf("Waiting for ack...\n");
+    kprintf("Waiting for ack (5s timeout)...\n");
+    u64 current_time, start_time = gettime();
+
     while ((data != PC_READY) && (data != PC_OK))
-        usb_recvbuffer_safe(channel, &data, 1);
+    {
+        current_time = gettime();
+
+        if (diff_sec(start_time, current_time) >= 5) {
```
Comment (9ary) [inline source/main.c:266]: nit: mixed brace style, but tbf it's a mess as it is anyway so it's fine

File: source/main.c [hunk 2]
```diff
@@ -256,9 +256,21 @@ int load_usb(char slot)
...
 
-    kprintf("Waiting for ack...\n");
+    kprintf("Waiting for ack (5s timeout)...\n");
+    u64 current_time, start_time = gettime();
+
     while ((data != PC_READY) && (data != PC_OK))
-        usb_recvbuffer_safe(channel, &data, 1);
+    {
+        current_time = gettime();
+
```
Comment (9ary) [inline source/main.c:265]: either remove the blank line (this is directly relevant to the if condition below) or fold the `gettime()` call into it, that shouldn't hurt readability imo

**Review by 9ary — APPROVED** [review]
Looks good to me, please address the style issues and we're good to merge.

Also the commit message feels a bit weird, how about this: "USB Gecko: give up after 5s timeout"
I feel like that would be a little more descriptive.

**Comment by webhdx** [comment]
Oh yeah I definitely had a brain fart when I was writing that title lol Fixed in [d8b043b](https://github.com/redolution/iplboot/pull/38/commits/d8b043be962529bcff17a76c39bb9395b7ca4379)

**Pull request merged**